src/teamplay.c

```diff
@@ -713,20 +764,38 @@ static void TeamplayReportTaken(gedict_t *client)
 		}
 		else if ((NEED(needFlags, it_health) || NEED(needFlags, it_armor) || NEED(needFlags, it_rl)
 				|| NEED(needFlags, it_lg) || NEED(needFlags, it_rockets)
-				|| NEED(needFlags, it_cells)) && HAVE_POWERUP(client))
+				|| NEED(needFlags, it_cells)))
 		{
 			// Note that we check if you are holding powerup. This is because TOOK remembers for 15 seconds.
 			// So a case could arise where you took quad then died less than 15 seconds later, and you'd be reporting "team need %u" (because $colored_powerups would be empty)
 			strlcpy(message, "{&c0b0team&cfff} ", sizeof(message));
-			strlcat(message, PowerupText(client), sizeof(message));
-			strlcat(message, " need ", sizeof(message));
+			if (HAVE_POWERUP(client))
+			{
+				strlcat(message, PowerupText(client), sizeof(message));
+				strlcat(message, " ", sizeof(message));
+			}
+			if (isCTF() && (HAVE_FLAG(client) || HAVE_RUNE(client)))
+			{
+				strlcat(message, CTFItemText(client), sizeof(message));
+				strlcat(message, " ", sizeof(message));
+			}
+			strlcat(message, "need ", sizeof(message));
```

Comment:
Everything here is unrelated to bots? Just additional tp messaging. Split to separate PR to trim down this big PR, can be merged on its own.